core/src/main/scala/pipeflow/dsl/actions/Closure.scala

```diff
@@ -0,0 +1,14 @@
+package pipeflow.dsl.actions
+
+import pipeflow.dsl.actions.Action.Evidence
+import pipeflow.dsl.actions.Closure.ClosureBlock
+
+object Closure {
+  type ClosureBlock = () => Unit
```

Comment:
Could it return something in order to be able to add code actions depending on the errors or even to "pipe" the output of one task to the next one?

Reply:
A `Closure` `Action` is just something to start playing with trivial actions, and still requires a lot of refinement, for example, for what should be the input parameters of the closure. Nevertheless the main reason to keep the action a part from the Task object is to favour composition instead of inheritance (so this is why I don't follow exactly how airflow defines the operators). Anything related to the execution of the task is responsibility of the Action and the Task has nothing to do with it.
The plan is to have different type of actions: Spark, SparkSubmit, DockerRun, ...
The Action specialisations will be pure declarations, they won't contain any logic for execution, but the actors that get materialised from them.
The idea of concatenating action results from different tasks sound very interesting, but I don't see how would it fit in this model.